File: csharp/main/ClassOrInterfaceDef.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary
{
	public abstract class ClassOrInterfaceDef<TProperty, TType> : CompositeTypeScope<TProperty, TType>
		where TType : Type
	{
		protected ClassOrInterfaceDef([NotNull] String name, [NotNull] ITypeScope parent, [CanBeNull, ItemNotNull] List<String> comments)
			: base(name, parent, comments)
		{
			SuperInterfaces = new List<InterfaceDef>();
		}

		[NotNull, ItemNotNull]
		public List<InterfaceDef> SuperInterfaces { get; }
	}
}

File: java/src/main/java/deltix/luminary/ClassOrInterfaceDef.java
package deltix.luminary;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

public abstract class ClassOrInterfaceDef<TProperty extends PropertyDef<? extends ClassOrInterfaceDef>, TType extends Type> extends CompositeTypeScope<TProperty, TType> implements TypeScope {
    private final List<InterfaceDef> superInterfaces = new ArrayList<>();

    public ClassOrInterfaceDef(@NotNull TypeScope parent, @NotNull String name, @Nullable List<String> comments) {
        super(parent, name, comments);
    }

    /**
     * List of super-interfaces inherited by this class or interface.
     *
     * @return List of super-interfaces.
     */
    @NotNull
    public List<InterfaceDef> getSuperInterfaces() {
        return superInterfaces;
    }
}
